### DISCUSSION TITLE
Qiita CLI で publish --force 'ベースファイル名' を実施すると対象以外のファイルも更新される事がある

### DISCUSSION THREAD

**Discussion by tettekete** [increments/qiita-discussions#946] [要望・アイデア・不具合]
## そもそもの問題

そもそもの経緯として、公開済みで編集も行っていない記事についてGitHub Actions 上で「内容がQiita上の記事より古い可能性があります」エラーが発生（条件不明）し、他のファイルの更新や公開ができなくなる問題があります。

状態の確認のためまずローカル環境で `npx qiita pull` を行いますが diff は発生しません。

## 対処法としての `publish --force`

逆にローカルから `publish` することで Qiita 側を更新する方法を取ると解決出来るため、`qiita publish --force 'ベースファイル名'` を行います。

_※ ちなみに `--force` 無しだとローカルでも「内容がQiita上の記事より古い可能性があります」エラーになります_

すると`'ベースファイル名'` 記事の `updated_at` が更新されローカルファイルに差分が発生します。

ところが、`'ベースファイル名'`  **以外**の記事の `updated_at` も更新される事があるようです（全ての記事ではないので、これも条件不明です）。

この差分を `git commit` ,`git push` しない限り「内容がQiita上の記事より古い可能性があります」エラーが解消されないため、更新を受け入れざるを得ません。

## 修正のお願い

このような状況で Qiita CLI を利用していきますと、記事の内容に変更はないのに更新日時だけが次々と更新されていくことになります。

原因の究明と修正をお願いいたします。

**Comment by acronhub** [comment]
@tettekete 
#945 でお聞きしている内容をもとに、併せて確認させていただきます 🙏

**Reply by acronhub** [reply]
@tettekete 
調査を進めたところ、
qiita publish <ベースファイル名> コマンドについて、指定した記事ファイルのみが対象になっていることを確認しました。

調査の結果、可能性があると思われる原因をみつけましたので、ご報告いたします。

以下の2つの条件を満たしている場合に、GitHub Actions のジョブが実行されるたびに限定共有記事が常に更新されてしまう状態になりました。
- ユーザー設定ファイル(qiita.config.json)で `includePrivate`を `true` に設定していない
- 記事ファイルに限定共有記事 `private: true` が含まれている

今後この挙動に関しては改善を検討いたします。
以上の条件に当てはまる場合は、一旦以下の対応を行うことで、常に限定共有記事を更新してしまう現象は回避できます。
ユーザー設定ファイル(qiita.config.json)の `includePrivate: true` に変更する
